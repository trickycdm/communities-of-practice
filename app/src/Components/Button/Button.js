import React, { memo } from 'react'
import PropTypes from 'prop-types'

const Button = memo(({ type = 'button', onClick, label = 'no label set...' }) => {
  return (
    <div className={`rounded-button__container`}>
      <button className={'rounded-button'} type={type} onClick={onClick}>{label}</button>
    </div>
  )
})

Button.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string,
  onClick: PropTypes.func
}

export default Button
