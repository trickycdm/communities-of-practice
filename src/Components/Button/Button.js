import styles from './Button.module.css'
import React, { memo } from 'react'
import PropTypes from 'prop-types'

console.log(styles)
const Button = ({ id, type = 'button', onClick, label = 'no label set...' }) => {
  return (
    <div className={styles['button-container']}>
      <button id={id} className={styles.button} type={type} onClick={onClick}>{label}</button>
    </div>
  )
}

Button.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string,
  onClick: PropTypes.func
}

// names and memo'd export to allow proper unit testing with enzyme https://github.com/airbnb/enzyme/issues/1875
export { Button }
export default memo(Button)
