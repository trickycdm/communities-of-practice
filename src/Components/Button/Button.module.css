.button {
    margin: 40px;
    background-color: green;
    border: 5px dashed pink;
}