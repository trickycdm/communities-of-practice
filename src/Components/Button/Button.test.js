import React from 'react';
import { Button } from './Button';
import { render, shallow } from 'enzyme';

const mockClickFunc = jest.fn();

describe('Button', () => {
  // simple render without any interactivity
  it('should render correctly with default props', () => {
    const component = render(<Button />);
    expect(component).toMatchSnapshot();
  });
  // mount the full component to check interactivity
  it('should render correctly with custom props', () => {
    const component = shallow(<Button id="test" label="Custom label" onClick={mockClickFunc} type="button" />);
    component.find('button#test').simulate('click');

    expect(component).toMatchSnapshot();
    expect(mockClickFunc).toHaveBeenCalled();
  });
});
