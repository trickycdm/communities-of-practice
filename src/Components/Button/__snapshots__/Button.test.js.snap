// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly with custom props 1`] = `ShallowWrapper {}`;

exports[`Button should render correctly with default props 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "button-container",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "button",
          "type": "button",
        },
        "children": Array [
          Object {
            "data": "no label set...",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "type": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "type": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
